source/core/coverage/ut_coverage.pkb

```diff
@@ -65,68 +65,104 @@ create or replace package body ut_coverage is
   begin
 
     if not ut_coverage_helper.is_develop_mode() then
       l_skipped_objects := ut_utils.get_utplsql_objects_list() multiset union set(g_skipped_objects);
     end if;
 
-    -- TODO - add inclusive and exclusive filtering
-    select ut_coverage_row(
-             lower(s.owner||'.'||s.name),
-             s.owner, s.name, s.type, s.line,
-           --filtering out false - negatives reported by profiler (zero executions while it should be ignored).
-           case when
-              regexp_instr(
-                s.text,
-                '^\s*(((not)?\s*(overriding|final|instantiable)\s*)*(constructor|member)?\s*(procedure|function)|begin|end\s*;)', 1, 1, 0, 'i'
-              ) = 0 then u.total_occur
-           end
-          )
-      bulk collect into l_data
+    --prepare global temp table with sources
+    delete from ut_coverage_sources_tmp;
+
+    insert into ut_coverage_sources_tmp(owner,name,line,text)
+    select s.owner,s.name,s.line,s.text
       from all_source s
-      --need to use owner privileges here
-      left join table( ut_coverage_helper.get_raw_coverage_data() ) u
-        on s.owner = u.owner
-       and s.name  = u.name
-       and s.type  = u.type
-       and s.line  = nvl(u.line_number, s.line)
      where s.type not in ('PACKAGE', 'TYPE')
        and s.owner in (select t.column_value from table(g_schema_names) t)
        --Exclude calls to utPLSQL framework and Unit Test packages
-       and not exists(select 1 from table(l_skipped_objects) l where s.owner = l.owner AND s.name = l.name)
-     order by s.owner, s.name, s.line;
-
-    for i in 1 .. l_data.count loop
+    --   and not exists(select 1 from table(l_skipped_objects) l where s.owner = l.owner AND s.name = l.name)
+    ;
+
+    for src_object in (
+      select o.owner, o.object_name, o.object_type, lower(o.owner||'.'||o.object_name) full_name, c.lines_count
+      from all_objects o
+      join (select max(c.line) lines_count, c.owner, c.name
+        from ut_coverage_sources_tmp c
+        group by c.owner, c.name) c
+        on o.owner = c.owner and o.object_name = c.name
+      where o.object_type not in ('PACKAGE', 'TYPE')
+      and o.owner in ( select t.column_value from table (g_schema_names) t)
+        --Exclude calls to utPLSQL framework and Unit Test packages
+      and not exists ( select 1 from table (l_skipped_objects) l where o.owner = l.owner and o.object_name = l.name)
+    ) loop
+
+      --get coverage data
+      l_line_calls := ut_coverage_helper.get_raw_coverage_data( src_object.owner, src_object.object_name );
+
+      --if there is coverage, we need to filter out the garbage (badly indicated data from dbms_profiler)
+      if l_line_calls.count > 0 then
+        --get source lines to skip
+        select line
+          bulk collect into l_source_lines
+          from ut_coverage_sources_tmp c
+         where c.owner = src_object.owner
+           and c.name = src_object.object_name
+           and regexp_instr(
```

Comment:
Do you think we should filter out while populating `ut_coverage_sources_tmp`?
I think you the use of `regexp_like` is more clear then `regexp_instr() != 0`